elastic/kibana
# [Fleet] Install the Fleet Server package during setup

## Summary

That PR add the creation of Fleet Server indices during setup, this is done by:
* installing the `fleet-server` package that create index template for `.fleet-*`
* creating the indexes if the indexes do not exists (this could be removed if https://github.com/elastic/package-spec/issues/117 is implemented)

The migration from saved object to fleet system indices has been moved from the plugin start to the fleet setup call as we need the indices to be created before.

Not done in that PR and will be in a followup PR: 
* we need an API guard that prevent the user to call Fleet APIs until the setup is done.

## How to test that PR?

* add this to your Kibana config `xpack.fleet.agents.fleetServerEnabled: true`
* Start kibana you should see the following in the logs `[warning][fleet][plugins] Extra setup is needed to be able to use central management for agent, please visit the Fleet app in Kibana.`
* Browse fleet and setup central management for agents
* If you create enrollment API keys you should see them in the index `.fleet-enrollment-api-keys`
